src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -522,6 +522,11 @@ private static void DeserializeFrameworkTypeValue(CodeWriter writer, CodeWriterD
                 writer.Append($"({frameworkType}){element}.GetString()");
                 return;
             }
+            else if (frameworkType.Namespace?.StartsWith("Azure.ResourceManager") == true)
+            {
+                writer.Append($"{frameworkType}.Deserialize{frameworkType.Name}({element})");
+                return;
+            }
```

Comment:
When matching some Resources models for Core package, we will not use PropertyReferenceType (which is themselves) for them and as they can find the same type in Core, they will be treated as frameworkType. Need to add this deserialize logic for them.

Further thinking: We should probably rethink the definition of frameworktype, should we treat those classes in `Azure.Resourcemanager` as frameworktypes?